src/backend/marsha/bbb/tests/test_api.py

```diff
@@ -0,0 +1,921 @@
+        )
+        self.assertEqual(response.status_code, 200)
+        new_uuid = re.search(
+            "http://testserver/lti/meetings/(.*)", response.json().get("new_url", "")
+        ).group(1)
+        self.assertEqual(uuid.UUID(new_uuid).version, 4)
+        self.assertDictEqual(
+            {
+                "new_url": f"http://testserver/lti/meetings/{new_uuid}",
+                "meetings": [
+                    {
+                        "id": str(meeting.id),
+                        "lti_id": str(meeting.lti_id),
+                        "lti_url": f"http://testserver/lti/meetings/{str(meeting.id)}",
+                        "title": meeting.title,
+                        "meeting_id": str(meeting.meeting_id),
+                        "playlist": {
+                            "id": str(meeting.playlist_id),
+                            "title": meeting.playlist.title,
+                            "lti_id": meeting.playlist.lti_id,
+                        },
+                    }
+                ],
+            },
+            response.json(),
+        )
+
+    @mock.patch.object(requests, "get")
+    def test_api_start_anonymous(self, mock_create_request: MagicMock):
+        """An anonymous should not be able to start a meeting."""
+        meeting = MeetingFactory()
+
+        response = self.client.patch(f"/api/meetings/{meeting.id}/start/")
+        self.assertEqual(response.status_code, 401)
+        mock_create_request.assert_not_called()
+
+    @mock.patch.object(requests, "get")
+    def test_api_start_user_logged_in(self, mock_create_request: MagicMock):
+        """A logged in user should not be able to start a meeting."""
+        user = core_factories.UserFactory(
+            first_name="Jane", last_name="Doe", email="jane.doe@example.com"
+        )
+        meeting = MeetingFactory()
+        self.client.force_login(user)
+
+        response = self.client.patch(f"/api/meetings/{meeting.id}/start/")
+        self.assertEqual(response.status_code, 401)
+        mock_create_request.assert_not_called()
+
+    @mock.patch.object(requests, "get")
+    def test_api_start_student(self, mock_create_request: MagicMock):
+        """A student should not be able to start a meeting."""
+        meeting = MeetingFactory()
+
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(meeting.id)
+        jwt_token.payload["roles"] = ["student"]
+        jwt_token.payload["permissions"] = {"can_update": True}
+
+        response = self.client.patch(
+            f"/api/meetings/{meeting.id}/start/",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+        self.assertEqual(response.status_code, 403)
+        mock_create_request.assert_not_called()
+
+    @mock.patch.object(requests, "get")
+    def test_api_start_new_meeting(self, mock_create_request: MagicMock):
+        """A new meeting should be started if no meeting exists for the same id."""
```

Comment:
I don't understand the description. You create a meeting using the `MeetingFactory`, you patch it. So it exists.

Reply:
I changed the description, but it's related to meeting existence in BBB side.

Reply:
Oh ok, We have the same word for a meeting in marsha and a meeting in BBB :laughing: Not easy to distinguish both